src/components/Form/ToggleButton.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+import { HtmlButton } from '../../reset';
+import { ToggleProps, baseClassName, ToggleState } from './Toggle';
+import { logger } from '../../utils/logger';
+
+export class ToggleButton extends Component<ToggleProps> {
+  state: ToggleState = {
+    toggleState: !!this.props.checked || !!this.props.defaultChecked || false,
+  };
+
+  componentWillReceiveProps(nextProps: ToggleProps) {
+    const { checked } = nextProps;
+    if (!!checked) {
+      this.setState({ toggleState: !!checked });
+    }
+  }
+
+  handleClick = () => {
+    const { checked, onClick } = this.props;
+    const { toggleState } = this.state;
+    if (checked === undefined) {
+      this.setState({ toggleState: !toggleState });
+    }
+    if (!!onClick) {
+      onClick({ toggleState: !toggleState });
+    }
+  };
+
+  render() {
+    const {
+      disabled = false,
+      children,
+      checked: dissMissChecked,
+      defaultChecked: dissMissDefaultChecked,
+      onClick: dissMissOnClick,
+      toggleInputProps,
+      toggleInputComponent,
+      ...passProps
+    } = this.props;
+    const { toggleState } = this.state;
+    const toggleButtonClassName = `${baseClassName}_button`;
+
+    if (!!toggleInputProps || !!toggleInputComponent) {
+      logger.error(
+        `ToggleButton does not utilize 'toggleInputProps' and 'toggleInputComponent' props.`,
+      );
+    }
+
+    return (
+      <HtmlButton
+        className={toggleButtonClassName}
+        aria-disabled={disabled}
+        disabled={disabled}
```

Comment:
These might not work with IE as expected unless something is forcing a complete re-render of the component. IE/Edge do not handle boolean attributes always properly as was discovered in non-related PR earlier. If an attribute remains in DOM when set to false (e.g. <div disabled="" >...</div>) an easy fix is to add a key and bind the value to what ever needs to trigger the "re-render".

Reply:
I tried with Win 10 and Edge and when I added `disabled={true}` it added `disabled=""` to the DOM.
But when I changed the value to `disabled={false}` the `disabled=""` was gone from the DOM.

So I did not make change regarding this for now as it seems to work.